axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java

```diff
@@ -498,27 +499,38 @@ public Timesheet getCurrentTimesheet() {
   }
 
   @Override
-  public Timesheet getCurrentOrCreateTimesheet() {
+  public Timesheet getCurrentOrCreateTimesheet() throws AxelorException {
     Timesheet timesheet = getCurrentTimesheet();
     if (timesheet == null) {
       timesheet =
           createTimesheet(
-              AuthUtils.getUser(), appHumanResourceService.getTodayDateTime().toLocalDate(), null);
+              AuthUtils.getUser().getEmployee(),
+              appHumanResourceService.getTodayDateTime().toLocalDate(),
+              null);
     }
     return timesheet;
   }
 
   @Override
-  public Timesheet createTimesheet(User user, LocalDate fromDate, LocalDate toDate) {
+  public Timesheet createTimesheet(Employee employee, LocalDate fromDate, LocalDate toDate)
+      throws AxelorException {
+    if (employee == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.LEAVE_USER_EMPLOYEE),
+          AuthUtils.getUser().getName());
```

Comment:
The user corresponding to the employee will not always be the current user, please remove this check and this message.